ejb3/src/main/java/org/jboss/as/ejb3/subsystem/deployment/AbstractEJBComponentResourceDefinition.java

```diff
@@ -51,11 +51,35 @@
  */
 public abstract class AbstractEJBComponentResourceDefinition extends SimpleResourceDefinition {
 
-    public static final SimpleAttributeDefinition COMPONENT_CLASS_NAME = new SimpleAttributeDefinitionBuilder("component-class-name", ModelType.STRING)
+     static final SimpleAttributeDefinition COMPONENT_CLASS_NAME = new SimpleAttributeDefinitionBuilder("component-class-name", ModelType.STRING)
             .setValidator(new StringLengthValidator(1))
             .setFlags(AttributeAccess.Flag.STORAGE_RUNTIME)
             .build();
 
+     static final SimpleAttributeDefinition JNDI_NAMES = new SimpleAttributeDefinitionBuilder("jndi-names", ModelType.LIST)
```

Comment:
@chengfang For all the various LIST attributes you should used one of the ListAttributeDefinition impls. My guess is all of them are essentially List<String>, so use StringListAttributeDefinition. The pattern would be

```
static final SimpleAttributeDefinition JNDI_NAMES = StringListAttributeDefinition.Builder.of("jndi-names")     
```

followed by what you've got now.

This results in the management API description having the correct 'value-type' as discussed in https://docs.wildfly.org/19.1/Admin_Guide.html#description-of-an-attribute